include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -203,10 +206,12 @@ class WidthQualifiedTrait
 template <typename ConcreteType, typename ParentType>
 class WidthQualifiedType
     : public FIRRTLType::TypeBase<ConcreteType, ParentType,
-                                  detail::WidthTypeStorage> {
+                                  detail::WidthTypeStorage,
+                                  circt::hw::FieldIDTypeInterface::Trait> {
```

Comment:
Can't help but notice that WidthQualifiedType and types derived from it don't have the `WidthQualifiedTrait`, :upside_down_face: .  Just amusing of all the types to lack it....

Anyway, I'm not sure there's an issue with that, other than maybe if you check via `hasTrait<..>` or something-- since it's not an interface to be cast to/accessed through.

Reply:
Yes... this could use some cleanup. The main thing would be to get more of the types over into ODS. Only the ODS types are using this `WidthQualifiedTrait`. 🥲